sdk/core/src/tasks/sboms/github/sync.rs

```diff
@@ -0,0 +1,409 @@
+use crate::entities::sboms::{SbomProviderKind};
+use crate::Error;
+
+use crate::entities::tasks::Task;
+use crate::entities::xrefs::{Xref, XrefKind};
+use crate::services::sboms::SbomService;
+use crate::tasks::TaskProvider;
+use async_trait::async_trait;
+use platform::persistence::mongodb::{Service, Store};
+use std::collections::HashMap;
+use std::process::Command;
+use std::sync::Arc;
+use serde_json::{json, Value};
+use crate::services::github::client::Repo;
+use crate::services::github::mongo::GitHubProviderMongoService;
+use crate::services::github::service::GitHubService;
+
+const CYCLONEDX_JSON_FORMAT: &str = "cyclonedx-json";
+
+// scheme:type/namespace/name@version?qualifiers#subpath
+fn generate_stopgap_purl(full_name: String, commit_hash: String) -> String {
+    format!("pkg:harbor:{}@{}", full_name, commit_hash)
+}
+
+/// Synchronizes SBOMS for a GitHub Group with Harbor.
+#[derive(Debug)]
+pub struct SyncTask {
+    mongo_svc: GitHubProviderMongoService,
```

Comment:
I think this will become a `RepositoryService`.  Just adding guidestones to go with earlier comments.

Reply:
Noted.  I added to my backlog